src/Component/AutoMapper/MapperContext.php

```diff
@@ -235,6 +235,8 @@ public static function withNewContext(array $context, string $attribute): array
 
         if (\is_array($context[self::ALLOWED_ATTRIBUTES][$attribute] ?? false)) {
             $context[self::ALLOWED_ATTRIBUTES] = $context[self::ALLOWED_ATTRIBUTES][$attribute];
+        } else {
+            unset($context[self::ALLOWED_ATTRIBUTES]);
```

Comment:
when creating the context for a child node, if "allowed attributes" is not find, this surely means all its children property are allowed.

other wise we keep the allow properties for level N in level N+1 which is problematic and maps empty objects